### PR TITLE
TMA-836: release brick takes deprecated objects into account

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/lcm/actions/collect_ldm_objects.rb [hunk 1]
```diff
@@ -28,7 +28,10 @@ def call(params)
           synchronize = params.synchronize.pmap do |info|
             from = info.from
             from_project = development_client.projects(from) || fail("Invalid 'from' project specified - '#{from}'")
-            objects = (from_project.attributes.to_a + from_project.labels.to_a + from_project.datasets.to_a + from_project.facts.to_a).map(&:uri)
+            # TMA-836 - reading objects from datasets includes deprecated ones
+            attributes = from_project.datasets.map(&:attribute_uris).flatten
```
Comment (yenkins) [inline lib/gooddata/lcm/actions/collect_ldm_objects.rb]: Calls 'from_project.datasets' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/actions/collect_ldm_objects.rb]: Refers to 'from_project' more than self (maybe move it to another class?) (FeatureEnvy)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/755/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/615/ : SUCCESS in 2m 04s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/876/ : SUCCESS in 1m 54s

File: lib/gooddata/lcm/actions/collect_ldm_objects.rb (hunk 1 above)
Comment (panjan) [reply lib/gooddata/lcm/actions/collect_ldm_objects.rb]: ⬆️ Good one. `from_project.datasets` makes an HTTP call.

File: spec/integration/models/label_spec.rb [hunk 2]
```diff
@@ -37,4 +37,13 @@
       end
     end
   end
+
+  describe '#deprecated=' do
+    it 'sets the label as deprecated' do
+      @label.meta['deprecated'] = 1
+      @label.save
+      @label = @project.attributes('label.csv_policies.customer')
```
Comment (panjan) [inline spec/integration/models/label_spec.rb]: There is `Label#reload!`

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/757/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/616/ : SUCCESS in 2m 11s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/877/ : SUCCESS in 2m 10s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/758/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/618/ : SUCCESS in 38s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/879/ : SUCCESS in 39s

**Comment by panjan** [comment]
ok to merge

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/285/ : SUCCESS in 21s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/619/ : SUCCESS in 2m 02s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/880/ : SUCCESS in 2m 12s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1213/ : FAILURE in 1h 15m 25s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/972/ : SUCCESS in 54m 22s

**Comment by panjan** [comment]
The failure of https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1213/ is unrelated. Ok to merge.

**Pull request merged**